app/src/main/java/de/westnordost/streetcomplete/quests/shop_type/CheckShopType.kt

```diff
@@ -43,10 +43,7 @@ class CheckShopType : OsmElementQuestType<ShopTypeAnswer> {
     override fun createForm() = ShopTypeForm()
 
     override fun applyAnswerTo(answer: ShopTypeAnswer, changes: StringMapChangesBuilder) {
-        val otherCheckDateKeys = LAST_CHECK_DATE_KEYS.filterNot { it == SURVEY_MARK_KEY }
-        for (otherCheckDateKey in otherCheckDateKeys) {
-            changes.deleteIfExists(otherCheckDateKey)
-        }
+        changes.deleteOtherCheckDates()
         when (answer) {
             is IsShopVacant -> {
                 changes.addOrModify(SURVEY_MARK_KEY, LocalDate.now().toCheckDateString())
```

Comment:
This could just become:
```suggestion
                changes.updateCheckDate()
```

Although it would then try deleting the other keys twice, although that could be solved by putting a deleteCheckDates in the is ShopType bit.

Reply:
No, it's fine like it is now.

Reply:
I'd like to create a function that encompasses this bit at least (even if the delete other is kept elsewhere):
`changes.addOrModify(SURVEY_MARK_KEY, LocalDate.now().toCheckDateString())`

Unless you really object, as that will continue to simplify the TagInfo JSON stuff (as I don't need to understand that I have to ignore LocalDate.now()).

Reply:
IIRC that executing this code

```
changes.deleteOtherCheckDates()
changes.updateCheckDate()
```

will result in an Exception, because the changes builder verifies that each change is only done once. This is why I wrote to leave it as it is.

Reply:
I've added a addOrModifyCheckDate function and switched to that to simplify the TagInfo JSON stuff.